app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java

```diff
@@ -0,0 +1,260 @@
+
+                }
+            }
+
+            @Override
+            public void onAccuracyChanged(Sensor sensor, int accuracy) {
+            }
+        }, sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER), SensorManager.SENSOR_DELAY_NORMAL);
+
+        FloatingActionButton mButtonTakePhoto = findViewById(R.id.take_photo_button);
+        mButtonTakePhoto.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                SharedPreferences prefs = getSharedPreferences("prefs", MODE_PRIVATE);
+                SharedPreferences.Editor edit = prefs.edit();
+                edit.putString("text", null);
+                edit.putString("imageDataPath", null);
+                edit.apply();
+                takePhoto();
+            }
+        });
+    }
+
+    /**
+     * Takes a photo, saves it inside internal storage and resets the last extracted text
+     *
+     * @modify SharedPreferences
+     * @author Romanello Stefano - modified by Leonardo Rossi and Balzan Pietro
+     */
+    private void takePhoto() {
+        cameraKitView.captureImage(new CameraKitView.ImageCallback() {
+            @Override
+            public void onImage(CameraKitView cameraKitView, final byte[] photo) {
+
+                Bitmap bitmapImage = BitmapFactory.decodeByteArray(photo, 0, photo.length, null);
+
+                //Image rotation
+
+                if(orientationResult != null)
+                {
+                    switch (orientationResult)
+                    {
+                        case "LR": bitmapImage=rotateImage(bitmapImage,90); break;
+                        case "LL": bitmapImage=rotateImage(bitmapImage,270); break;
+                        case "PU": bitmapImage=rotateImage(bitmapImage,180); break;
+                        default: break;
+                    }
+
+                    //Temporary stores the captured photo into a file that will be used from the Camera Result activity
+                    String filePath= tempFileImage(CameraActivity.this, bitmapImage,"capturedImage");
+
+                    SharedPreferences prefs = getSharedPreferences("prefs", MODE_PRIVATE);
+                    SharedPreferences.Editor edit = prefs.edit();
+                    edit.putString("imagePath", filePath.trim());
+                    edit.apply();
+
+                    // analyze the brightness   @author Balzan Pietro
+                    final Bitmap bitmapImage2 = bitmapImage;
+                    new Thread(new Runnable() {      // run the brightness recognition on a new thread to improve performance
+                        public void run() {
+                            Context appContext= getApplicationContext();
+                            BrightnessRecognition bRecog = new BrightnessRecognition(bitmapImage2, appContext, handler);
+                            bRecog.imgBrightness(190,80,5);
+                        }
+                    }).start();
+
```

Comment:
I can see no error in the code, the style document is respected and the comment are enough (also because i read the class building). I would like to know if this method can compromise the speed of execution. Overall it's ok.

Reply:
When it wasn't on a separate thread the "takePhoto" method would wait for imgBrightness to finish its task before calling the intent to open the ResultActivity, so it did affect performance (there was a second where the program would just freeze) but running it on the new thread solved the problem